docs/src/implementation.md

```diff
@@ -1,16 +1,22 @@
 # Implementation details
 
-`CategoricalArray` is based on the `CategoricalPool` type, which keeps track of the levels and associates them with an integer reference (for internal use). It offers methods to set levels, change their order while preserving the references, and efficiently get the integer index corresponding to a level and vice-versa. It is also parameterized on the type used to store the references, so that small pools can use as little memory as possible. Finally, it keeps a vector of value objects (`CategoricalValue`), so that `getindex` can return the existing object instead of allocating a new one.
+`CategoricalArray` is made of the two fields:
 
-`CategoricalArray` is made of two fields:
+- `refs`: an integer array that stores the position of the category level in the `index` field of `CategoricalPool` for each `CategoricalArray` element; `0` denotes a missing value (for `CategoricalArray{Union{T, Null}}` only).
+- `pool`: the `CategoricalPool` object that maintains the levels of the array.
 
-- `refs`: an integer vector giving the index of the level in the pool for each element; if the array is nullable, `0` indicates a missing value.
-- `pool`: the `CategoricalPool` object keeping the levels of the array.
+The `CategoricalPool{V,R,C}` type keeps track of the levels of type `V` and associates them with an integer reference code of type `R` (for internal use). It offers methods to set the levels, change their order while preserving the references, and efficiently get the integer index corresponding to a level and vice-versa. Whether the values of `CategoricalArray` are ordered or not is defined by an `ordered` field of the pool. Finally, `CategoricalPool` keeps a `valindex` vector of value objects of type `C` (`CategoricalString{R}` or `CategoricalValue{V, R}`), so that `getindex` can return the existing object instead of allocating a new one.
 
-Whether an array (and its values) are ordered or not is stored as a property of the pool.
+The type parameters of `CategoricalArray{T, N, R <: Integer, V, C, U}` are a bit complex:
+ - `T` is the effective type of array elements unwrapped from `CategoricalString`/`CategoricalValue` helpers; if `T >: Null`, then the array is nullable.
```

Comment:
"effective" isn't really different from "actual". Better not use any adjective. BTW, "helpers"  would better be "wrappers", though it's redundant with "unwrapped", which could be changed to "without" or something like that.

Reply:
From google translate, one of the meanings: *"Effective: fulfilling a specified function in fact, though not formally acknowledged as such"* I think it's fits pretty well, but I can remove if you disagree.

I can either just delete "helpers" or "stripped of CatString/CatValue wrappers" or "without CatString/CatValue wrappers". Which one do you like? :)

Reply:
That definition doesn't really apply AFAIK, since indexing into the array will return categorical values, so the retrieved elements do not act as `T` elements for that "specified function".

Regarding the second point, "without" sounds simpler to me.

Reply:
The *"specified function"* here IMO is the efficient storage of `T` elements, while *formally* indexing returns categorical values. But I'm not bound to that adjective, will remove; "without CatString/CatValue wrappers" is already clear enough.